source/code/plugins/oms_common.rb

```diff
@@ -478,6 +480,141 @@ class Common
     @@tzRightFolder = 'right/'
 
     class << self
+
+      # Internal methods
+      # (left public for easy testing, though protected may be better later)
+
+      def clean_hostname_string(hnBuffer)
+        s = hnBuffer.strip
+        # Take off any port numbers:
+        if /\A(\h{4}:\h{4}:\h{4}:\h{4}:\h{4}:\h{4}:\h{4}:\h{4}):\d+\z/ =~ s
+            return $1
+        elsif not is_like_ipv6_string?(hnBuffer) and /\A(.*):\d+\z/ =~ s
+            return $1
+        end
+        return s
+      end
+
+      def has_Designated_HostnameFile?
+        return false if @@HostnameFilePath.nil?
+        return false if @@HostnameFilePath.length < 8 # 8 is first Guess.  Need policy here.
+        File.exist?(@@HostnameFilePath)
+      end
+
+      def is_dot_separated_string?(hnBuffer)
+        return true if /[^.]+\.[^.]+/ =~ hnBuffer
+        return false
+      end
+
+      def is_hostname_compliant?(hnBuffer)
+        # RFC 2181:
+        #   Size limit is 1 to 63 octets, so probably bytesize is appropriate method.
+        return false if hnBuffer.nil?
+        return false if /\./ =~ hnBuffer # Hostname by definition may not contain a dot.
+        return false if /:/ =~ hnBuffer # Hostname by definition may not contain a colon.
+        return true if 1 <= hnBuffer.bytesize && hnBuffer.bytesize <= 63
+        return false
+      end
+
+      def is_like_ipv4_string?(hnBuffer)
+        return false unless /\A[12]?\d{1,2}\.[12]?\d{1,2}\.[12]?\d{1,2}\.[12]?\d{1,2}\z/ =~ hnBuffer
+        qwa = hnBuffer.split('.')
+        return false unless qwa.length == 4
+        i = 0
+        qwa.each do |quadwordstring|
+            bi = quadwordstring.to_i
+            # This may need more detail if 255 octets are sometimes allowed, but I don't think so.
+            return false unless 0 <= bi and bi < 255
+            return false if i == 0 and bi == 0
+            i += 1
+        end
+        return false unless qwa[0].to_i != 0
+        return true
+      end
+
+      def is_like_ipv6_string?(hnBuffer)
+        return true if /\A\h{4}:\h{4}:\h{4}:\h{4}:\h{4}:\h{4}:\h{4}:\h{4}\z/ =~ hnBuffer
+        return false
+      end
+
+      def is_recognized_prefix_label?(fqdnLabel)
+        s = fqdnLabel.strip.downcase
+        return true if s == 'www'
+        return false
+      end
+
+      def look_for_sockety_host_address
```

Comment:
nit: sockety?

Reply:
I don't mind "socket_like"  or "Ruby_Socket_Class_hostname", but was in the short run thinking sockety might be adequately illustrative.  The main thing of interest that I want to convey here is that this is the hostname information that is typical from the Ruby Socket class as limited by our parameters used.  I see this as helpful because we want to be clear for test and usage maintenance activities.

Reply:
Makes sense. I just wasn't sure if it was a typo, or if you intended to convey "socket-like"